docker/registry/internal/base_client.go

```diff
@@ -195,6 +189,10 @@ func (c baseClient) url(pathTemplate string, args ...interface{}) string {
 
 // Ping pings the baseClient endpoint.
 func (c baseClient) Ping() error {
+	if !c.repoDetails.IsPrivate() {
```

Comment:
It seems unclear why IsPrivate() is correlated with api version 1 or 2.
Should the APIVersion() method be used instead?

Reply:
Ping here was used for credential validation which is why here skipped for non private. Also some of v1 root endpoint always requires credentials.
But now we always uses v2 which supports anonymous login. 
I am just removing this check here to allow Ping for both public and private cases.
There is one special case, gitlab root endpoint still does not work even for anonymous login. 
So I will skip ping for gitlab only.